lib/fibonacci.rb

```diff
@@ -1,8 +1,19 @@
 # Improved Fibonacci
 
-# Time Complexity - ?
-# Space Complexity - ?  (should be O(n))
-# Hint, you may want a recursive helper method
+# Time Complexity - O(n)
+# Space Complexity - O(n), but could we get this to O(1) using a hash table somehow?
 def fibonacci(n)
```

Comment:
This works, but doesn't have very efficient space complexity.  Could you solve this by only keeping track of the last 2 Fibonacci numbers.